### PR TITLE
Re-org the historian directory to publish historian as package

### PR DESCRIPTION
**Pull request by Sapphire1211**
1. Created two packages: historian and historian-base, moved most part of historian to historian-base, leave only www.ts under historian;
2. Refactored www.ts, moved most code into runner.ts, runnerFactory.ts

### CODE REVIEW

**Comment by curtisman** [comment]
Close and reopening to trigger the pipeline

File: server/historian/packages/historian-base/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# historian
```
Comment (tanviraumi) [inline server/historian/packages/historian-base/README.md:1]: This file should remain inside historian folder.

File: server/historian/packages/historian/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# historian
+
```
Comment (tanviraumi) [inline server/historian/packages/historian/README.md:2]: Follow the routerlicious and routerlicious-base package for readme's. And move this one to top level folder.

**Review by tanviraumi — COMMENTED** [review]
Did not review the files that are just copied. I don't know why git did not track them as moved.

Also the build pipeline errors need to be fixed.

File: server/historian/lerna.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "packages": [
```
[the remark this answers is not in the document]
Comment (tanviraumi) [reply server/historian/lerna.json:2]: Yes. Same as routerlicious

File: server/historian/packages/historian-base/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# @fluidframework/historian-base
+
+The historian base package exports base r11s constructs consumable by service providers.
```
Comment (tanviraumi) [inline server/historian/packages/historian-base/README.md:3]: Why r11s?

File: server/historian/packages/historian/package.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    "eslint": "eslint --ext=ts,tsx --format stylish src",
+    "eslint:fix": "eslint --ext=ts,tsx --format stylish src --fix",
+    "lint": "npm run eslint",
+    "lint:fix": "npm run eslint:fix",
+    "start": "node dist/www.js",
+    "tsc": "tsc",
+    "tslint": "tslint 'src/**/*.ts'"
+  },
+  "dependencies": {
+    "@fluidframework/gitresources": "^0.1012.0",
```
Comment (tanviraumi) [inline server/historian/packages/historian/package.json:23]: You dont need this right?

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**